### ISSUE TITLE
Mage2gen should abandon AbstractExtensibleObject class in favor of AbstractExtensibleModel AND lose the seperate DTO model

### ISSUE THREAD

**Issue by peterkeijsers** [krukas/Mage2Gen#278]
Mage2gen uses a depricated class for buidling it's extensible models.
It needs to extend AbstractExtensibleModel instead of AbstractExtensibleObject.

Magento also choose not to use DTO models anymore in the majority of its modules. All the DTO logic that was in Model/Data/dataModel.php should be moved to the Model/model.php class. We can also create an extra boolean option in Mage2gen to create a seperate DTO model when the user wants to.

Example below of a combined Model and DTO model which extends the AbstractExtensibleModel class:

**New Model class**

```
<?php

namespace Experius\Model\Model;

use Experius\Model\Model\ResourceModel\Model as ResourceModel;
use Magento\Framework\Model\AbstractExtensibleModel;
use Experius\Api\Api\Data\ExampleModelInterface;

class Model extends \Magento\Framework\Model\AbstractModel
{
    protected function _construct()
    {
        $this->_init(ResourceModel::class);
    }

    // All get and set methods from the DTO model should be put here

/**
     * Get entity_id
     * @return string|null
     */
    public function getEntityId()
    {
        return $this->getData(self::ENTITY_ID);
    }

    /**
     * Set entity_id
     * @param string $examplemodelId
     * @return \Experius\Api\Api\Data\ExampleModelInterface
     */
    public function setEntityId($entityId)
    {
        return $this->setData(self::ENTITY_ID, $entityId);
    }
}

```

**Comment by evs-xsarus** [comment]
Has this been added?

**Comment by Alainvdz99** [comment]
Hi @lewisvoncken, I am wondering why this issue is feature request is closed? The Magento 2.4(BETA) code generation is still not up to date with the `AbstractExtensibleModel ` which replaces the deprecated `AbstractExtensibleObject`. I am also agreeing @peterkeijsers that Magento doesn't use DTO's anymore. If you are agreeing with me, could you consider open this issue again?

Thanks
